source/common/config/grpc_mux_impl.cc

```diff
@@ -14,6 +14,24 @@
 namespace Envoy {
 namespace Config {
 
+namespace {
+struct AllMuxesState {
+  void insert(GrpcMuxImpl* mux) {
+    absl::WriterMutexLock locker(&lock_);
+    muxes_.insert(mux);
+  }
+
+  void erase(GrpcMuxImpl* mux) {
+    absl::WriterMutexLock locker(&lock_);
+    muxes_.erase(mux);
+  }
+
+  absl::flat_hash_set<GrpcMuxImpl*> muxes_;
+  absl::Mutex lock_;
```

Comment:
Would it be possible to make these members private and expose a shutdownAll method instead?